### PR TITLE
Gdpr retreive personal data from mod inbox

### PR DESCRIPTION
**Pull request by aleklisi**
This PR allows to retrieve personal data form `mod_inbox` backends.

### CODE REVIEW

File: big_tests/tests/gdpr_SUITE.erl [hunk 1]
```diff
@@ -113,7 +113,21 @@ end_per_testcase(CN, Config) ->
     escalus:end_per_testcase(CN, Config).
 
 inbox_required_modules() ->
-    [{mod_inbox, []}].
+    [
+     {mod_muc_light, [{host, binary_to_list(muclight_domain())},
```
Comment (ludwikbukowski) [inline big_tests/tests/gdpr_SUITE.erl]: why is it really added?

File: big_tests/tests/gdpr_SUITE.erl [hunk 2]
```diff
@@ -240,14 +254,17 @@ retrieve_private_xml(Config) ->
...
+            ExpectedHeader = ["jid", "content", "unread_count", "timestamp"],
             ExpectedItems = [
-                             #{ "content" => Body, "jid" => BobJid }
+                             #{ "content" => [{contains, Body}],
+                                "jid" => [{contains, BobS},
+                                          {contains, BobU}],
+                                "unread_count" => "1" }
                             ],
             retrieve_and_validate_personal_data(
               Alice, Config, "inbox", ExpectedHeader, ExpectedItems)
```
Comment (ludwikbukowski) [inline big_tests/tests/gdpr_SUITE.erl]: Can we assert the inbox of `Bob` as well?

**Comment by ludwikbukowski** [comment]
What do you think for testing scenarios like retrieve inbox for muclight or retrieve inbox for different values of `unread count`?

File: src/inbox/mod_inbox.erl [hunk 3]
```diff
@@ -78,6 +78,33 @@
 
 -export_type([get_inbox_params/0]).
 
+%%--------------------------------------------------------------------
+%% gdpr callbacks
+%%--------------------------------------------------------------------
+
+-behaviour(gdpr).
```
Comment (fenek) [inline src/inbox/mod_inbox.erl]: It breaks the convention where we have behaviour and exports at the top of the file.

File: src/inbox/mod_inbox.erl [hunk 4]
```diff
@@ -78,6 +78,33 @@
...
+    LUser = jid:nodeprep(Username),
+    LServer = jid:nameprep(Server),
+    Schema = ["jid", "content", "unread_count", "timestamp"],
+    InboxParams = #{
+        start => {0,0,0},
+        'end' => erlang:timestamp(),
+        order => asc,
+        hidden_read => false
+       },
+    Record = mod_inbox_backend:get_inbox(LUser, LServer, InboxParams),
```
Comment (fenek) [inline src/inbox/mod_inbox.erl]: Definitely not "record". Either "records" or even better - "entries".

File: big_tests/tests/gdpr_SUITE.erl [hunk 5]
```diff
@@ -290,11 +307,11 @@ retrieve_and_validate_personal_data(Alice, Config, FilePrefix, ExpectedHeader, E
              })
     end.
 
-csv_to_maps(ExpectedHeader, [HeaderRow | [Rows]]) ->
+csv_to_maps(ExpectedHeader, [ExpectedHeader | Rows]) ->
```
Comment (fenek) [inline big_tests/tests/gdpr_SUITE.erl]: Such fixes should be handled separately, or even directly in base branch, as it affects the logic of all other branches.

File: big_tests/tests/gdpr_SUITE.erl (hunk 1 above)
Comment (aleklisi) [reply big_tests/tests/gdpr_SUITE.erl]: works without so I'll remove it

File: big_tests/tests/gdpr_SUITE.erl (hunk 2 above)
Comment (aleklisi) [reply big_tests/tests/gdpr_SUITE.erl]: yes, we can

File: src/inbox/mod_inbox.erl (hunk 3 above)
Comment (aleklisi) [reply src/inbox/mod_inbox.erl]: I'll move to the top of the file.

File: src/inbox/mod_inbox.erl (hunk 4 above)
Comment (aleklisi) [reply src/inbox/mod_inbox.erl]: I'll rename this variable to `Entries`

**Pull request closed without merging**